spec/compiler/parser/warnings_spec.cr

```diff
@@ -1,8 +1,10 @@
 require "../../support/syntax"
 
+private TEST_FILENAME = "/foo/bar/baz.cr"
```

Comment:
What's the point of adding this constant (and using a longer value)?
IMO the expectations are more readable with a hard-coded `test.cr` instead of `#{TEST_FILENAME}`.

Reply:
This is the same placeholder filename used in `parser_spec.cr`.

Also, if `./test.cr` exists locally, then all the warning / error texts will currently pick up its contents to produce the source lines (this check is done by `File.file?` [here](https://github.com/crystal-lang/crystal/blob/bf0a606f5355645757f27cdec18de478a289cbe4/src/compiler/crystal/exception.cr#L90)), and the warnings specs will fail. `/foo/bar/baz.cr` has a far lower chance of existing locally.

Reply:
I have kept the name in `parser_spec.cr` because it was used there to test the behavior of `__FILE__` and `__DIR__` too